ui/src/app/workspace-app-routing.tsx

```diff
@@ -0,0 +1,228 @@
+import {CohortPage} from 'app/cohort-search/cohort-page/cohort-page.component';
+import {AppRoute, withRouteData} from 'app/components/app-router';
+import * as React from 'react';
+import {NOTEBOOK_PAGE_KEY} from './components/help-sidebar';
+import {InteractiveNotebook} from './pages/analysis/interactive-notebook';
+import {NotebookList} from './pages/analysis/notebook-list';
+import {NotebookRedirect} from './pages/analysis/notebook-redirect';
+import {CohortReview} from './pages/data/cohort-review/cohort-review';
+import {DetailPage} from './pages/data/cohort-review/detail-page';
+import {QueryReport} from './pages/data/cohort-review/query-report.component';
+import {ParticipantsTable} from './pages/data/cohort-review/table-page';
+import {CohortActions} from './pages/data/cohort/cohort-actions';
+import {ConceptHomepage} from './pages/data/concept/concept-homepage';
+import {ConceptSearch} from './pages/data/concept/concept-search';
+import {ConceptSetActions} from './pages/data/concept/concept-set-actions';
+import {DataComponent} from './pages/data/data-component';
+import {DatasetPage} from './pages/data/data-set/dataset-page';
+import {WorkspaceAbout} from './pages/workspace/workspace-about';
+import {WorkspaceEdit, WorkspaceEditMode} from './pages/workspace/workspace-edit';
+import {BreadcrumbType} from './utils/navigation';
+
+const CohortPagePage = withRouteData(CohortPage);
+const CohortActionsPage = withRouteData(CohortActions);
+const CohortReviewPage = withRouteData(CohortReview);
+const ConceptHomepagePage = withRouteData(ConceptHomepage);
+const ConceptSearchPage = withRouteData(ConceptSearch);
+const ConceptSetActionsPage = withRouteData(ConceptSetActions);
+const DataComponentPage = withRouteData(DataComponent);
+const DataSetComponentPage = withRouteData(DatasetPage);
+const DetailPagePage = withRouteData(DetailPage);
+const InteractiveNotebookPage = withRouteData(InteractiveNotebook);
+const NotebookListPage = withRouteData(NotebookList);
+const NotebookRedirectPage = withRouteData(NotebookRedirect);
+const ParticipantsTablePage = withRouteData(ParticipantsTable);
+const QueryReportPage = withRouteData(QueryReport);
+const WorkspaceAboutPage = withRouteData(WorkspaceAbout);
+const WorkspaceEditPage = withRouteData(WorkspaceEdit);
+
+// TODO angular2react: Adding memo here feels a little off but it was necessary to prevent workspace-wrapper from
+// rendering over and over again on page load, rendering (hah) the app unusable.
+// We should be able to refactor this once we are driving the entire app through React router.
+export const WorkspaceRoutes = React.memo(() => {
```

Comment:
We should definitely refactor this ASAP, on the next step or after conversion. I think this function has too many caveats

Reply:
agreed